lib/rubocop/cop/rspec/factory_bot/create_list.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Prefer using create_list over n.times { create :obj } calls.
+        #
+        # @example
+        #   # bad
+        #   3.times { create :user }
+        #
+        #   # good
+        #   create_list :user, 3
+        #
+        #   # good
+        #   3.times { |n| create :user, created_at: n.months.ago }
+        class CreateList < Cop
+          MSG = 'Prefer create_list.'.freeze
+
+          def_node_matcher :n_times, '(send (int $_) :times)'
+
+          def_node_matcher :times_block_without_args?, <<-PATTERN
+            (block
+              #n_times
+              (args)
+              ...
+            )
+          PATTERN
+
+          def_node_matcher :factory_call, <<-PATTERN
+            (send ${(const nil? {:FactoryGirl :FactoryBot}) nil?} :create (sym $_) $...)
+          PATTERN
+
+          def on_block(node)
+            return unless times_block_without_args?(node)
+
+            receiver, _args, body = *node
+
+            return unless factory_call(body)
+
+            add_offense(receiver, location: :expression)
+          end
+
+          def autocorrect(node)
+            block = node.parent
+            replacement = generate_replacement(block)
+            lambda do |corrector|
+              corrector.replace(block.loc.expression, replacement)
+            end
+          end
+
+          private
+
+          def generate_replacement(block)
+            receiver, _args, body = *block
+            count = n_times(receiver)
+            factory_call_replacement(body, count)
+          end
+
+          def method_uses_parens?(node)
```

Comment:
Since it’s only called from `#format_method_call`, I’d prefer to have this method definition at the bottom of the class.

Reply:
No longer used